File: Java/Easy/Accepted/largest-substring-between-two-equal-characters_8.java
class Solution {
    public int maxLengthBetweenEqualCharacters(String s) {
        
        int[] a = new int[26];
        Arrays.fill(a, -1);
        int max = -1;
        for (int i = 0; i < s.length(); i++) {
            int j = s.charAt(i) - 'a';
            if (a[j] < 0) {
                a[j] = i;
            } else {
                max = Math.max(max, i - a[j] - 1);
            }
        }
        return max;
    }
}

File: C++/Easy/Accepted/largest-substring-between-two-equal-characters_8.cpp
class Solution {
public:
    int maxLengthBetweenEqualCharacters(string s) {
        
        int n = s.size();
        int maxLen = -1;
        for(int i = 0; i < n; i++){
            for(int j = n - 1; j > i; j--){
                if(s[i] == s[j]){
                    maxLen = max(maxLen, j - i - 1);
                    break;
                }
            }
        }
        return maxLen;
    }
};